File: src/main/java/com/murtukov/css_to_jss/PasteAsJssAction.java
package com.murtukov.css_to_jss;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.command.WriteCommandAction;
import com.intellij.openapi.ide.CopyPasteManager;
import org.jetbrains.annotations.NotNull;

import java.awt.datatransfer.DataFlavor;

public class PasteAsJssAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent event) {
        var cpManager = CopyPasteManager.getInstance();
        var contents = cpManager.getContents();
        var project = event.getProject();
        var editor = event.getData(CommonDataKeys.EDITOR);
        var document = editor.getDocument();
        var primaryCaret = editor.getCaretModel().getPrimaryCaret();

        assert contents != null;

        var parser = new CssConverter();

        WriteCommandAction.runWriteCommandAction(project, () -> {
            try {
                document.replaceString(
                    primaryCaret.getSelectionStart(),
                    primaryCaret.getSelectionEnd(),
                    parser.parse(contents.getTransferData(DataFlavor.stringFlavor).toString())
                );
            } catch (Exception e) {
                e.printStackTrace();
            }
        });

        primaryCaret.removeSelection();
    }

    @Override
    public boolean isDumbAware() {
        return true;
    }
}

File: src/main/kotlin/org/murtukov/css_to_jss/PasteAsJssAction.kt
package org.murtukov.css_to_jss

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ide.CopyPasteManager
import com.intellij.openapi.actionSystem.CommonDataKeys
import com.intellij.psi.codeStyle.CodeStyleManager
import com.intellij.psi.PsiDocumentManager
import com.intellij.openapi.command.WriteCommandAction
import java.awt.datatransfer.DataFlavor
import java.lang.Exception

class PasteAsJssAction : AnAction() {
    override fun actionPerformed(event: AnActionEvent) {
        val cpManager     = CopyPasteManager.getInstance()
        val contents      = cpManager.contents
        val project       = event.project
        val editor        = event.getData(CommonDataKeys.EDITOR)
        val psiFile       = event.getData(CommonDataKeys.PSI_FILE)
        val document      = editor!!.document
        val primaryCaret  = editor.caretModel.primaryCaret
        val csManager     = CodeStyleManager.getInstance(project!!)
        val psiDocManager = PsiDocumentManager.getInstance(project)
        val converter     = CssConverter()

        WriteCommandAction.runWriteCommandAction(project) {
            val start = primaryCaret.selectionStart
            val end = primaryCaret.selectionEnd
            try {
                val string = converter.convert(contents!!.getTransferData(DataFlavor.stringFlavor).toString())
                document.replaceString(start, end, string)
                psiDocManager.commitDocument(document)
                csManager.reformatText(psiFile!!, start, start + string.length)
            } catch (ignored: Exception) {}
        }
    }

    override fun isDumbAware(): Boolean {
        return true
    }
}